cdap-ui/app/cdap/components/Wrangler/__tests__/type-inference.test.js

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+import {inferType, inferColumn} from 'components/Wrangler/type-inference';
+
+
+describe('Wrangler: Type Inference', () => {
+  it('should infer string', () => {
+    const input1 = 'test 123';
+    const input2 = '123 test';
+
+    expect(inferType(input1)).toBe('string');
+    expect(inferType(input2)).toBe('string');
+  });
+
+  it('should infer booleans', () => {
+    const trueInput = 'true';
+    const falseInput = 'false';
+
+    expect(inferType(trueInput)).toBe('boolean');
+    expect(inferType(falseInput)).toBe('boolean');
+  });
+
+  it('should infer integer', () => {
+    const integerInput = '135';
+    const wrongInteger = '08';
```

Comment:
08` incorrect but `8` is fine? As a user if I have 0s padded to the left of number then I would definitely assume that it will be ignored. Not sure if this is what we want as per the product.

Also I am not sure if I understand the `parseFloat` usage here to mark this as invalid number. Isn't `parseFloat(08)` returns `8` which we can infer as number? (as `8%1 === 0`) which is still correct right?